### PR TITLE
docs(firestore): update Android & iOS manual integration steps

### PR DESCRIPTION
**Pull request by katharinepadilha**
### Summary

This PR updates the firestore documentation for v6.

- Add iOS documentation
- Add android documentation

### Checklist

- [ ] Supports `Android`
- [ ] Supports `iOS`
- [ ] `e2e` tests added or updated in [/tests/e2e/\*](/tests/e2e)
- [ ] Updated the documentation in the [docs repo](https://github.com/invertase/react-native-firebase-docs)
  - **LINK TO DOCS PR HERE**
- [ ] Flow types updated
- [ ] Typescript types updated

### Release Plan

[CATEGORY][type] [LOCATION] - Message

---

Think `react-native-firebase` is great? Please consider supporting the project with any of the below:

- 👉 Donate via [Open Collective](https://opencollective.com/react-native-firebase/donate)
- 👉 Follow [`React Native Firebase`](https://twitter.com/rnfirebase) and [`Invertase`](https://twitter.com/invertaseio) on Twitter
- 👉 Star this repo on GitHub ⭐️
- 👉 Contribute; see our [contributing guide](/CONTRIBUTING.md)

### CODE REVIEW

File: docs/firestore/android.md [hunk 1]
```diff
@@ -1,6 +1,63 @@
...
+    return Arrays.asList(
+      new MainReactPackage(),
+      new ReactNativeFirebaseFirestorePackage(),
+      // ..
+```
+
+#### Enable multidex
+
+Adding Firestore to your Android app requires [`multiDexEnabled` to be set to `true`](https://developer.android.com/studio/build/multidex) in `android/app/build.gradle`:
+
```
Comment (mikehardy) [inline docs/firestore/android.md]: There should probably be a note here about adding the androidx.multidex dependency and altering your MainApplication to extend MultidexApplication as well like these 3 lines https://github.com/mikehardy/rnfbdemo/blob/master/make-demo.sh#L127

**Comment by mikehardy** [comment]
Thanks for helping with the docs! These all seem correct, my only question is if they are all needed for manual linking or if some of them (or all of them?) apply even in auto-linking situations. I know at least the Multidex thing is needed whether you auto-link or not

File: docs/firestore/android.md (hunk 1 above)
Comment (Ehesp) [reply docs/firestore/android.md]: Also, this isn't included when auto-linking. Do you think we should be adding it automatically for the user?

Comment (mikehardy) [reply docs/firestore/android.md]: That seems touchy, I believe the object hierarchy in someone's MainApplication is not deterministic. You could do a heuristic but you can't know for sure. Additionally some people may have decided to drop support for API<21? I dunno

Comment (Ehesp) [reply docs/firestore/android.md]: Yeah, my thinking is we just hook onto the dex error and output a custom error message.

File: docs/firestore/android.md [hunk 2]
```diff
@@ -1,6 +1,63 @@
 ---
-title: Android Setup
-description: Manually integrate Cloud Firestore into your Android application.
+title: Firestore Android Initialization
+description: Firestore initialization with Android.
 ---
 
 # Android Manual Linking
+
+> The following steps are only required if your environment does not have access to React Native
```
Comment (Ehesp) [inline docs/firestore/android.md]: Can we change this to:

```
The following steps are only required if you are using React Native <= 0.59 or need to manually integrate the library.
```

**Comment by Ehesp** [comment]
cc @Phaen ^^

File: docs/firestore/android.md (hunk 2 above)
Comment (katharinepadilha) [reply docs/firestore/android.md]: Sure! Updated on android.md and ios.md 😄

File: docs/firestore/android.md (hunk 1 above)
Comment (katharinepadilha) [reply docs/firestore/android.md]: Added the missing multidex setup! 😄

Comment (mikehardy) [reply docs/firestore/android.md]: Awesome :-)

File: docs/firestore/android.md [hunk 3]
```diff
@@ -1,56 +1,91 @@
 ---
-title: Android Setup
-description: Manually integrate Cloud Firestore into your Android application.
+title: Firestore Android Initialization
+description: Firestore initialization with Android.
```
Comment (Salakar) [inline docs/firestore/android.md]:
```suggestion
description: Manually integrate Cloud Firestore into your Android application.
```

File: docs/firestore/android.md [hunk 4]
```diff
@@ -1,56 +1,91 @@
 ---
-title: Android Setup
+title: Firestore Android Initialization
```
Comment (Salakar) [inline docs/firestore/android.md]:
```suggestion
title: Cloud Firestore Android Integration
```

File: docs/firestore/ios.md [hunk 5]
```diff
@@ -1,12 +1,11 @@
 ---
-title: iOS Setup
-description: Manually integrate Cloud Firestore into your iOS application.
+title: Firestore iOS Initialization
```
Comment (Salakar) [inline docs/firestore/ios.md]:
```suggestion
title: Cloud Firestore iOS Integration
```

File: docs/firestore/ios.md [hunk 6]
```diff
@@ -1,12 +1,11 @@
 ---
-title: iOS Setup
-description: Manually integrate Cloud Firestore into your iOS application.
+title: Cloud Firestore iOS Integration
+description: Firestore initialization with iOS.
```
Comment (Salakar) [inline docs/firestore/ios.md]:
```suggestion
description: Manually integrate Cloud Firestore into your iOS application.
```

File: docs/firestore/ios.md [hunk 7]
```diff
@@ -1,12 +1,11 @@
...
-title: iOS Setup
+title: Cloud Firestore iOS Integration
 description: Manually integrate Cloud Firestore into your iOS application.
 ---
 
 # iOS Manual Linking
 
-> The following steps are only required if your environment does not have access to React Native
-> auto-linking.
+> The following steps are only required if your environment does not have access to React Native auto-linking.
```
Comment (Salakar) [inline docs/firestore/ios.md]:
```suggestion
> The following steps are only required if your environment does not have access to React Native auto-linking (<= 0.59) or you need to manually integrate the library.
```

File: docs/firestore/android.md [hunk 8]
```diff
@@ -1,56 +1,91 @@
...
-title: Android Setup
+title: Cloud Firestore Android Integration
 description: Manually integrate Cloud Firestore into your Android application.
 ---
 
-# Android Manual Installation
+# Android Manual Linking
 
-The following steps are only required if you are using React Native <= 0.59 or need to manually integrate the library.
+> The following steps are only required if you are using React Native without auto-linking (<= 0.59) or need to manually integrate the library.
```
Comment (Salakar) [inline docs/firestore/android.md]:
```suggestion
> The following steps are only required if you are using React Native without auto-linking (<= 0.59) or you need to manually integrate the library.
```

**Comment by Salakar** [comment]
@katharinepadilha hey, sorry it took so long to get this approved 🙈 - I've updated with master and made some small tweaks - am happy to merge so will go ahead and do that.

Thanks again for sending this over 🎉

**Pull request merged**